pandas/core/arrays/interval.py

```diff
@@ -176,6 +180,24 @@
 """
 
 
+def maybe_convert_numeric_to_64bit(arr: NumpyIndexT) -> NumpyIndexT:
+    # IntervalTree only supports 64 bit numpy array
+
+    if not is_array_like(arr):
+        return arr
```

Comment:
just for my understanding, what's an example of where this makes a difference?

Reply:
This was just a short-circuit, so the functions breaks early, if the value can't possibly be array-like, it made no funcional difference. This will overall probably not be an improvement as the `arr` inn the current version can't be non-array and the function is typed, so I can remove it again. 

I moved this function to `core.dtypes.cast` and renamed it `is_array_like`.

Reply:
Yeah would remove this if not necessary. This would get me guessing how to get here if I would want to make a change in a couple of weeks